spec.bs

```diff
@@ -1881,6 +1928,178 @@ Several APIs specific to fenced frames are defined on the {{Fence}} interface.
   </wpt>
 </div>
 
+<div algorithm>
+  The <dfn method for=Fence>disableUntrustedNetwork()</dfn> method steps are:
+
+  1. Let |p| be [=a new promise=].
+
+  1. Let |context| be [=this=]'s [=relevant global object=]'s [=Window/browsing context=].
+
+  1. If |context| is null, then throw a {{SecurityError}} {{DOMException}}.
+
+  1. Let |instance| be |context|'s [=browsing context/fenced frame config instance=].
+
+  1. If |instance| is null, then [=resolve=] |p| with {{undefined}} and return |p|.
+
+  1. If the [=relevant settings object=]'s [=environment settings object/origin=] and
+     |instance|'s [=fenced frame config instance/mapped url=]'s [=url/origin=] are not [=same
+     origin=], then [=reject=] |p| with a {{TypeError}}.
+
+  1. If [=this=]'s [=relevant global object=]'s [=Window/navigable=]'s [=navigable/traversable
+     navigable=] is not a [=fenced navigable container/fenced navigable=], then [=resolve=] |p| with
+     {{undefined}} and return |p|.
+
+  1. Let |global| be [=this=]'s [=relevant global object=].
+
+  1. Run the following steps [=in parallel=]:
+
+     1. Let |fencedFrameNonce| be |instance|'s [=fenced frame config instance/partition nonce=].
+
+     1. Let |credentiallessNonce| be
+
+     Issue: the page credentialless nonce
+     (<a href="https://github.com/WICG/fenced-frame/issues/191">WICG/fenced-frame#191</a>)
```

Comment:
Before we land this, I'd like https://github.com/WICG/fenced-frame/issues/191#issuecomment-2365349593 to be addressed so I can evaluate how much of a hole in the current spec this is. A rough design, or at least design thoughts (["concepts of a plan"](https://www.reddit.com/r/PoliticalHumor/comments/1fdzo9m/mr_trump_do_you_have_a_plan/#lightbox), if you will) will suffice.

Reply:
Added concepts of a plan to the issue. I think we can just grab the nonce itself as part of this PR rather than having a TODO. I'll need to move this issue comment to another part of the algorithm that will need a patch in a follow up.